google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java

```diff
@@ -272,7 +272,31 @@ static StandardTableDefinition fromPb(Table tablePb) {
       builder.setStreamingBuffer(StreamingBuffer.fromPb(tablePb.getStreamingBuffer()));
     }
     if (tablePb.getTimePartitioning() != null) {
-      builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      try {
+        builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            "Illegal Argument - Got unexpected time partitioning "
+                + tablePb.getTimePartitioning().getType()
+                + " in project "
+                + tablePb.getTableReference().getProjectId()
+                + " in dataset "
+                + tablePb.getTableReference().getDatasetId()
+                + " in table "
+                + tablePb.getTableReference().getTableId(),
+            e);
+      } catch (NullPointerException e) {
+        throw new NullPointerException(
+            "Null pointer - Got unexpected time partitioning "
+                + tablePb.getTimePartitioning().getType()
```

Comment:
Same here. Let's call getTimePartitioning().toPrettyString()

Reply:
changing to using toString() since toPrettyString calls for adding IOException to method signature